packages/cdk/lib/transcription-service.ts

```diff
@@ -714,6 +737,85 @@ export class TranscriptionService extends GuStack {
 		outputHandlerLambda.addToRolePolicy(getParametersPolicy);
 		outputHandlerLambda.addToRolePolicy(putMetricDataPolicy);
 
+		const webpageSnapshotQueue = new Queue(
+			this,
+			`${APP_NAME}-webpage-snapshot-queue`,
+			{
+				queueName: `${APP_NAME}-webpage-snapshot-queue-${this.stage}`,
+				// this needs to be greater than the timeout of the webpage snapshot lambda
+				visibilityTimeout: Duration.minutes(15),
+			},
+		);
+
+		addSubscription(
+			this,
+			`WebpageSnapshot`,
+			webpageSnapshotQueue,
+			combinedTaskTopic,
+		);
+
+		const chromiumLayerKey = new GuStringParameter(
+			this,
+			'ChromiumLayerZipKey',
+			{
+				description:
+					"Key for the chromium layer's zip file (pushed to layerBucket by publish-chromium-layer.sh script)",
+				default: 'chromium-v140.0.0-layer.arm64.zip',
+			},
+		);
+
+		const chromiumLayer = new LayerVersion(
+			this,
+			`ChromiumLayer_arm64-${this.stage}`,
+			{
+				code: Code.fromBucket(
+					lambdaLayerBucket,
+					chromiumLayerKey.valueAsString,
+				),
+				description: 'Chromium Layer',
+				layerVersionName: 'ChromiumLayer',
+				compatibleArchitectures: [Architecture.ARM_64],
+				compatibleRuntimes: [
+					Runtime.NODEJS_LATEST,
+					Runtime.NODEJS_22_X,
+					Runtime.NODEJS_20_X,
+					Runtime.NODEJS_18_X,
+				],
+			},
+		);
+
+		const webpageSnapshotLambda = new GuLambdaFunction(
+			this,
+			'transcription-service-webpage-snapshot',
+			{
+				fileName: 'webpage-snapshot.zip',
+				handler: 'index.webpageSnapshot',
+				runtime: Runtime.NODEJS_20_X,
+				architecture: Architecture.ARM_64,
+				timeout: Duration.seconds(900),
+				memorySize: 2048,
+				layers: [chromiumLayer],
+				app: `${APP_NAME}-webpage-snapshot`,
+				errorPercentageMonitoring:
+					this.stage === 'PROD'
+						? {
+								toleratedErrorPercentage: 0,
+								noMonitoring: false,
+								snsTopicName: alarmTopicName,
+							}
+						: undefined,
+			},
+		);
+
+		webpageSnapshotLambda.addEventSource(
+			new SqsEventSource(webpageSnapshotQueue),
+		);
+
+		webpageSnapshotLambda.addToRolePolicy(getParametersPolicy);
+		webpageSnapshotLambda.addToRolePolicy(putMetricDataPolicy);
+
+		webpageSnapshotLambda.addToRolePolicy(giantRemoteIngestOutputSendPolicy);
+
```

Comment:
minor: might be possible to add these in the config of the lambda itself, via `initialPolicy` array